### PR TITLE
[doc] Listing Pages Documentation

### PR DESCRIPTION
**Pull request by ccatherinetan**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
* added basic documentation of Listing Pages, i.e. descriptions of supabase calls and components used 
- `ListingPage`
    - `ListingCard` 
    - `ListingDetails`
        - `ProfileMatch`
        - `InterestForm`

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @kyrenetam at the bottom."

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- wasn't sure what `fullText` (parameter for DropdownInput) does 
- not sure if I explained usage of `onChange` for `ListingCard` correctly 
- maybe convert the bulleted instructions into numbered instructions + add numbering to example code 
- remove the "Frontend" and "Document Translation" etc. H3's, unless there's info to add there.  
- if we decide to stick with the Component-based structure, i can make the components all H2's so that they show up in the page's table of contents ("On This Page")

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'
- I've been using the notion guide: https://www.notion.so/calblueprint/IJP-Developer-Docs-48c79572f41c4b869ec05aba8198a301
- also referencing React documentation for layout/structure: https://react.dev/reference/react/useEffect
- https://www.tablesgenerator.com/markdown_tables
- styling tables https://stackoverflow.com/questions/36121672/set-table-column-width-via-markdown

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @varortz

### CODE REVIEW

**Comment by jinkang-0** [comment]
Looks great so far! Just some suggestions:
- It might be good to mention how to customize filters for each listing page in case the future dev teams want to do that
- I don't think we need to iterate about each filter type and listing details it displays, since that's more of a design concern than a dev one (rather, devs would probably want to know **how** to customize them if they're ever asked to do that)
- For components, I think it would be good to describe the props that they take so devs can get a sense of **how** to use those components and better understand what each component does intuitively

File: docs/dev/app/listings.md [hunk 1]
```diff
@@ -1,9 +1,388 @@
...
+}
+```
+
+**Fields of a Filter**
+- `id`: unique identifier of the filter's dropwdown button
+- `options`: All options for a given filter that a user can select from (derived from the listing data). This can be a map or a set. A map is used when you want the options' display-text to be different from the stored-text. For the map, keys are the actual values stored, and values are the displayed value
+- `placeholder`: The placeholder display text on the filter dropdown when no values of the current filter are selected. 
+- `value`: The selected values of the filter. This should be tracked as a state in the main component, and its corresponding setState function should be passed into `onChange` below.
+- `onChange`: Usually of the form `newValue => setFieldFilters(newValue)`. This function is called when the values of the filter dropdown is changed, i.e. when the user selects/deselects a value of a the current filter. 
+- (optional) `fullText` (!! not sure?)
```
Comment (ccatherinetan) [inline docs/dev/app/listings.md]: need to update this

File: docs/dev/app/listings.md [hunk 2]
```diff
@@ -1,9 +1,388 @@
...
+## Components
+**Components Used**
+- `ListingPage`
+    - `ListingCard` 
+    - `ListingDetails`
+        - `ProfileMatch`
+        - `InterestForm`
+
+### `ListingPage`
+- The listing pages are all instances of `ListingPage`. The ListingPage contains a scrollable list of ListingCards on the left, and the ListingDetails of the currently selected listing card on the right.
```
Comment (jinkang-0) [inline docs/dev/app/listings.md]: nit: `ListingPage` also includes filters, but that's a minor detail

File: docs/dev/app/listings.md [hunk 3]
```diff
@@ -1,9 +1,388 @@
...
+    );
+
+    // update languageOptions (filter options) based on listingData
+    const languageOptions = useMemo(
+        () => new Set(listingData.flatMap(listing => listing.languages)),
+        [listingData],
+    );
+
+    // update filteredListings based on filter options  
+    const filteredListings = useMemo( () => {...},
```
Comment (jinkang-0) [inline docs/dev/app/listings.md]: nit: can we have the arrow function be on its own line, as we had it for other `useMemo` hooks before this?

File: docs/dev/app/listings.md [hunk 4]
```diff
@@ -1,9 +1,388 @@
...
+    const filter1Options = useMemo(
+        () => {...} // get filter options from listingData 
+        [listingData],
+    );
+
+    // update languageOptions (filter options) based on listingData
+    const languageOptions = useMemo(
+        () => new Set(listingData.flatMap(listing => listing.languages)),
+        [listingData],
+    );
```
Comment (jinkang-0) [inline docs/dev/app/listings.md]: it may be beneficial to explain that the filter options can be a set or a map

File: docs/dev/app/listings.md [hunk 5]
```diff
@@ -1,9 +1,388 @@
...
+            filteredListings={filteredListings}
+            selectedListing={selectedListing}
+            setSelectedListing={lisitng => setSelectedListing(listing)}
+            interpretation
+        />
+    );
+}
+```
+
+**ListingPage Props**
```
Comment (jinkang-0) [inline docs/dev/app/listings.md:48]: personally i think it's better to have this first so readers can get a sense of what the component needs.
also this is getting a bit long - since we already have working examples of the `ListingPage` component in the codebase, we can just make a quick reference to that here instead of making up arbitrary code.

File: docs/dev/app/listings.md [hunk 6]
```diff
@@ -1,9 +1,388 @@
...
+- `filters`: *Filter[]*. A list of filters (see following section on customizing filters). Each filter in the list creates a new dropdown filter button. 
+- `resetFilters`: *() => void*. A no-argument function that clears all filters, triggered when "Reset Filters" is clicked. 
+- `filteredListings`: *Listing[]*. A list of listings, filtered by the currenlty selected filter options. This should be a memoized constant dependent on selected filters and listing data. 
+- `selectedListing`: *Listing | null*. The currently selected listing whose ListingDetails will be displayed and whose ListingCard will be highlighted. In the above example, the defualt `selectedListin` is the first item of the `filteredListings` if one exists. 
+- `setSelectedListing`: *(listing: Listing | null) => void*. Of the form `lisitng => setSelectedListing(listing)` as above, this function is used to set/update the selectedListing when it changes, i.e. when the user clicks on a new listing. 
+-  (optional)`interpretation`: *boolean*. Default value, false. A boolean specifying whether the current listing page is for interpretations or not. Used to differentiate between CaseInterpretations and Cases.
```
Comment (jinkang-0) [inline docs/dev/app/listings.md]: nit on the formatting - when rendered, the wall of text seem really jumbled and messy
image

it could be better if we had the prop name on its own line, along with the type, then explain what it does
for example:

`filters`: `Filter[]`

A list of filters (see section below to customize filters). Each filter...

File: docs/dev/app/listings.md [hunk 7]
```diff
@@ -1,9 +1,388 @@
...
+}
+```
+
+**Fields of a Filter**
+- `id`: unique identifier of the filter's dropwdown button
+- `options`: All options for a given filter that a user can select from (derived from the listing data). This can be a map or a set. A map is used when you want the options' display-text to be different from the stored-text. For the map, keys are the actual values stored, and values are the displayed value
+- `placeholder`: The placeholder display text on the filter dropdown when no values of the current filter are selected. 
+- `value`: The selected values of the filter. This should be tracked as a state in the main component, and its corresponding setState function should be passed into `onChange` below.
+- `onChange`: Usually of the form `newValue => setFieldFilters(newValue)`. This function is called when the values of the filter dropdown is changed, i.e. when the user selects/deselects a value of a the current filter. 
+- (optional) `fullText`: A string that is the default display name of the filter dropdown button. Currenlty, this is only used for the "Remote/In Person" filter.
```
Comment (jinkang-0) [inline docs/dev/app/listings.md]: similar note for the filter props: a bit too condensed, could reformat to take less horizontal real estate

also, since we're going over each prop here, we can probably get rid of the code block for the interface

File: docs/dev/app/listings.md [hunk 8]
```diff
@@ -1,9 +1,388 @@
...
+
+**Fields of a Filter**
+- `id`: unique identifier of the filter's dropwdown button
+- `options`: All options for a given filter that a user can select from (derived from the listing data). This can be a map or a set. A map is used when you want the options' display-text to be different from the stored-text. For the map, keys are the actual values stored, and values are the displayed value
+- `placeholder`: The placeholder display text on the filter dropdown when no values of the current filter are selected. 
+- `value`: The selected values of the filter. This should be tracked as a state in the main component, and its corresponding setState function should be passed into `onChange` below.
+- `onChange`: Usually of the form `newValue => setFieldFilters(newValue)`. This function is called when the values of the filter dropdown is changed, i.e. when the user selects/deselects a value of a the current filter. 
+- (optional) `fullText`: A string that is the default display name of the filter dropdown button. Currenlty, this is only used for the "Remote/In Person" filter. 
+
+### `ListingCard`
```
Comment (jinkang-0) [inline docs/dev/app/listings.md]: since we kind of made this up, it could be good to add an image here so they know what we're talking about

File: docs/dev/app/listings.md [hunk 9]
```diff
@@ -1,9 +1,388 @@
...
+
+**ListingCard Props**
+- `listing`: *Listing*. The listing whose ListingCard will be rendered. The rendering of fields is dependent on the `listing_type` of `listing`. 
+- (optional) `isSelected`: *boolean*. Default value: false. If true, the card will be highlighted. 
+- (optional) `onClick`: *(id: UUID) => void*. An additional event handler triggered when the ListingCard is clicked. 
+- (optional) `interpretation`: *boolean*. Default value: false. Should be `true` for Langauge Support listings. Used to distinguish between Case Interpretations and Cases. 
+
+**Example Usage**
+- Langauge Support listings: `<ListingCard listing={listingData} isSelected={isSelected} interpretation />`
+- Non-Langauge Support listings: `<ListingCard listing={listingData} isSelected={isSelected} />`
```
Comment (jinkang-0) [inline docs/dev/app/listings.md]: for code that wraps around like this, it's better to just use a code block (\`\`\`)

File: docs/dev/app/listings.md [hunk 10]
```diff
@@ -1,9 +1,388 @@
...
+
+### `ListingDetails`
+- ListingDetails displays the details of a selected listing. If the user is logged in and onboarded, the ProfileMatch and InterestForm appear. Otherwise, the Profilematch and InterestForm will be replaced by buttons directing the user to log in and complete onboarding.
+- To display the fields underneath "Highlights", an instance of the `ListingFields` component is used. For nullable fields, if a listing field is null/undefined, the field will appear as "Not Available." However, some fields are non-nullable.
+
+**ListingDetails Props**
```
Comment (jinkang-0) [inline docs/dev/app/listings.md:222]: similar crit for listing detail props

File: docs/dev/app/listings.md [hunk 11]
```diff
@@ -1,9 +1,388 @@
...
+- `listingData`: *Listing*. The listing whose ListingDetails will be rendered. The rendering of fields is dependent on the `listing_type` of `listingData`. 
+- (optional) `interpretation`: *boolean*. Default value: false. Should be `true` for Langauge Support listings. Used to distinguish between Case Interpretations and Cases. 
+
+**Example Usage**
+- Langauge Support listings: `<ListingDetails listingData={listingData} interpretation />`
+- Non-Langauge Support listings: `<ListingDetails listingData={listingData} />`
```
Comment (jinkang-0) [inline docs/dev/app/listings.md]: if we update the previous usage, it could be good to update this as well to be consistent

File: docs/dev/app/listings.md [hunk 12]
```diff
@@ -1,9 +1,388 @@
...
+            )
+        }
+        // ... handling of other listing types
+    }, [listingData, interpretation]);
+
+    // ... 
+}
+```
+
+#### Changing/Adding Icons
```
Comment (jinkang-0) [inline docs/dev/app/listings.md]: this can be helpful outside of listings as well, so we should probably put this in a separate "Miscellaneous" page under "Application"

feel free to start that!

File: docs/dev/app/listings.md [hunk 13]
```diff
@@ -1,9 +1,388 @@
...
+const newIcon = <Icon type="newIcon" />
+```
+
+### `ProfileMatch`
+- For a logged in and onboarded user, `ProfileMatch` indicates aspects of their profile meet the requirements of a listing.
+- `matchIcon` conditionally renders the match icon based on the value of the argument, `match`, of type `boolean | undefined`. If `match` is undefined, the `grayDot` icon is returned, indicating that the listing does not include information about a given field. 
+- `renderIconGroup` renders the icon and text for a match-field. It takes in the argument `fields` of type `MatchField`. However, for fields that require the `yellowExclaimation` icon, `renderIconGroup` is not used. 
+    - The `yellowExclaimation` icon indicates that the user's data does not match a listing's details, but it is not a strict requirement. In particular, `yellowExclaimation` is used when an attorney does not match the language for the case.
+
+**ProfileMatch Props**
```
Comment (jinkang-0) [inline docs/dev/app/listings.md:324]: similar crit for props format

File: docs/dev/app/listings.md [hunk 14]
```diff
@@ -1,9 +1,388 @@
...
+            {/* other match fields */}
+        </Flex>
+}
+```
+
+### `InterestForm`
+- For a logged in and onboarded user, `InterestForm` enables a user to submit an interest for the current listing. This interest is saved in the `interests` table. 
+- `handleInsert` is used to handle the insertion of the user's responses to the interest form into the `interests table`. After error-checking that all required fields (according to listing type) are valid and filled out, the user's responses are upserted. If the a user submits multiple forms for one case, only the most recent response will be saved on the `interests` table.
+
+**InterestForm Props**
```
Comment (jinkang-0) [inline docs/dev/app/listings.md:386]: similar crit for props and usage

File: docs/dev/app/listings.md [hunk 15]
```diff
@@ -1,9 +1,388 @@
...
+- Non-Langauge Support listings: `<InterestForm listingData={listingData} />`
+
+#### How to Customize InterestForm Questions 
+**Creating a New Question**
+1. Create a new Input Component and a new state to track the input. Update useEffect to reset the state when listingData changes. 
+2. Add a new field to the `Responses` type, corresponding to this new input 
+3. In `handleInsert`, add an error check if this new field is required for certain listing types 
+4. In `handleInsert`, conditionally add the new input to `responses` to be inserted into supabase
+
+```tsx
```
Comment (jinkang-0) [inline docs/dev/app/listings.md]: instead of this long chunk of code, it's better to ask the reader to reference the existing codebase and only include short, relevant code blocks - this makes it a lot easier for the reader to understand what to focus on and what to change, one step at a time

if everything must be noted, it's better to split it into separate chunks rather than a big block like this

File: docs/dev/app/listings.md [hunk 16]
```diff
@@ -1,9 +1,388 @@
...
+                <Flex $gap="30px" $direction="column">
+                    {/* (1) New Interest Question Input*/}
+                    {/*... Existing Questions*/}
+                </Flex>
+    )
+}
+```
 
 ## Cases
+### Frontend
```
Comment (jinkang-0) [inline docs/dev/app/listings.md]: no need to split frontend and backend
the framework we have is that we only talk about frontend (client side), and any backend logic should either be in the "Supabase" page or "Admin Dashboard" (renamed to "Retool" in `docs` branch)

Comment (jinkang-0) [reply docs/dev/app/listings.md]: here, you can just give the reader a brief description of how each specific listing page differ from one another, then split into categories only if necessary

File: docs/dev/app/listings.md (hunk 2 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: added!

File: docs/dev/app/listings.md (hunk 5 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md:48]: gotcha, i moved up the listingpage props, and i'll prob replace the example code w/ a reference to the LCA page. perhaps i'll leave some code demonstrating how to create a new filter?

File: docs/dev/app/listings.md (hunk 6 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: how's this? 
Screen Shot 2024-05-30 at 3 38 02 PM

File: docs/dev/app/listings.md (hunk 12 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: sg! added the misc page! 
do we also want to specify when to use webp vs. svg?

File: docs/dev/app/listings.md (hunk 11 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: sg, i've updated all Example Usage lines to be code blocks

File: docs/dev/app/listings.md (hunk 15 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: gotcha, i split up the code chunk

File: docs/dev/app/listings.md (hunk 4 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: gotcha, i ended up removing this section of code, but i mentioned this briefly when describing the fields of a filter.

File: docs/dev/app/listings.md (hunk 16 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: gotcha, i deleted the description of the query functions and added the comparison table at the beginning

File: docs/dev/app/listings.md [hunk 17]
```diff
@@ -1,9 +1,390 @@
...
+
+- `ListingPage`
+    - `ListingCard` 
+    - `ListingDetails`
+        - `ProfileMatch`
+        - `InterestForm`
+
+## `ListingPage`
+
+::: note Description
```
Comment (ccatherinetan) [inline docs/dev/app/listings.md]: do we like the note or the tip better? 
Screen Shot 2024-05-31 at 1 05 36 AM
Screen Shot 2024-05-31 at 1 05 26 AM

**Review by ccatherinetan — COMMENTED** [review]
I didn't mean to start a review, but here are some things i'd appreciate someone helping me out with 
- deciding whether to add to (or even keep) the listing comparison table
- tip vs. note for description of components 
- is the way i've split tutorial code blocks clear enough?

File: docs/dev/app/listings.md (hunk 12 above)
Comment (jinkang-0) [reply docs/dev/app/listings.md]: sure! my intuition is that if the source is a vector graphic, then it should be SVG; otherwise, any regular old image should be converted to webp

File: docs/dev/app/listings.md (hunk 17 above)
Comment (jinkang-0) [reply docs/dev/app/listings.md]: i think the top one looks better since it gives the block a bit more contrast

File: docs/dev/app/listings.md [hunk 18]
```diff
@@ -1,9 +1,390 @@
...
+<ListingCard listing={listingData} isSelected={isSelected} interpretation />
+```
+- Non-Langauge Support listings: 
+``` tsx
+<ListingCard listing={listingData} isSelected={isSelected} />
+```
+
+## `ListingDetails`
+::: note Description
+`ListingDetails` displays the details of a selected listing. If the user is logged in and onboarded, the ProfileMatch and InterestForm appear. Otherwise, the Profilematch and InterestForm will be replaced by buttons directing the user to log in and complete onboarding.
```
Comment (varortz) [inline docs/dev/app/listings.md]: could we update to "log in or complete onboarding" instead of using "and"?

File: docs/dev/app/listings.md (hunk 17 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: gotcha, changed!

File: docs/dev/app/listings.md [hunk 19]
```diff
@@ -1,9 +1,416 @@
...
+| Filters | <ol><li>Remote/In Person</li><li>Languages</li><li>Adjudicating Agency</li><li>Country of Origin</li></ol> | <ol><li>Country Field</li><li>Language(s)</li></ol> | <ol><li>Listing Type</li><li>Language(s)</li></ol> | <ol><li>Listing Type</li><li>Language(s)</li></ol> | <ol><li>Listing Type</li><li>Language(s)</li></ol> |
+| Listing Highlights | <ol><li>Relief sought</li><li>Time Commitment</li><li>Remote/In Person</li><li>Adjudicating Agency</li><li>Client Languages</li><li>Client Country of Origin</li><li>__*__ Client Location</li></ol> | <ol><li>Country Field</li><li>Language(s)</li><li>Expected Deliverable</li></ol> | **DOC** <ol><li>Language(s)</li><li>Number of Pages</li></ol> **CASE_INT** <ol><li>Language(s)</li><li>Time Commitment</li><li> Remote/In Person </li></ol> **INT** <ol><li>Language(s) </li><li>Remote/In Person</li></ol> |
+
+::: note (*) Note: 
+If the client is in custody (i.e., `is_detained`, pulled from LegalServer, is `true`), "Client Location" appears as "Custody Location." Otherwise, it remains as Client Location.
+::: 
+
+## **Components Used**
+
+- `ListingPage`
```
Comment (varortz) [inline docs/dev/app/listings.md]: Might be helpful to also list the `FilterDropdown` as one of the components!

File: docs/dev/app/listings.md [hunk 20]
```diff
@@ -1,9 +1,416 @@
...
+            onChange: newValue => setFilter1Filters(newValue),
+            placeholder: 'Filter1',
+        },
+        // other filters ... 
+    ]}
+    // other props 
+/>
+```
+
+## `ListingCard`
```
Comment (varortz) [inline docs/dev/app/listings.md:179]: WOWOWOW, thanks for adding info for all the components used in the `ListingPage` component!!!

File: docs/dev/app/misc.md [hunk 21]
```diff
@@ -0,0 +1,20 @@
+# Miscellaneous
```
Comment (varortz) [inline docs/dev/app/misc.md]: What if we renamed this page/file to "Styling"/"styling.md"? I feel like adding or changing icons could fall under styling, and we could also include info about global styles, text/font in here (maybe we can add these on a diff branch). Miscellaneous may be a bit too vague.

File: docs/dev/app/listings.md [hunk 22]
```diff
@@ -1,9 +1,416 @@
...
+```
+- Non-Langauge Support listings: 
+``` tsx:no-line-numbers
+<ProfileMatch listingData={listingData} />
+```
+
+### How to Customize ProfileMatch Fields
+
+The easiest way to create a new profile match field is to create a new constant of type `MatchField` as defined below. 
+```ts
```
Comment (varortz) [inline docs/dev/app/listings.md]: could we remove line numbers for all the code blocks?

File: docs/dev/app/listings.md [hunk 23]
```diff
@@ -1,9 +1,416 @@
...
+        },
+        // other filters ... 
+    ]}
+    // other props 
+/>
+```
+
+## `ListingCard` 
+
+::: tip Description
```
Comment (varortz) [inline docs/dev/app/listings.md]: Is it okay if we remove the tip containers from the all descriptions?

**Review by varortz — COMMENTED** [review]
THANK YOU SO MUCH FOR ALL THIS AMAZING DOCUMENTATION! Left a couple comments

File: docs/dev/app/listings.md [hunk 24]
```diff
@@ -1,9 +1,416 @@
...
+|:------------:|------|--------------------------|------------------|
+| Filters | <ol><li>Remote/In Person</li><li>Languages</li><li>Adjudicating Agency</li><li>Country of Origin</li></ol> | <ol><li>Country Field</li><li>Language(s)</li></ol> | <ol><li>Listing Type</li><li>Language(s)</li></ol> | <ol><li>Listing Type</li><li>Language(s)</li></ol> | <ol><li>Listing Type</li><li>Language(s)</li></ol> |
+| Listing Highlights | <ol><li>Relief sought</li><li>Time Commitment</li><li>Remote/In Person</li><li>Adjudicating Agency</li><li>Client Languages</li><li>Client Country of Origin</li><li>__*__ Client Location</li></ol> | <ol><li>Country Field</li><li>Language(s)</li><li>Expected Deliverable</li></ol> | **DOC** <ol><li>Language(s)</li><li>Number of Pages</li></ol> **CASE_INT** <ol><li>Language(s)</li><li>Time Commitment</li><li> Remote/In Person </li></ol> **INT** <ol><li>Language(s) </li><li>Remote/In Person</li></ol> |
+
+::: note (*) Note: 
+If the client is in custody (i.e., `is_detained`, pulled from LegalServer, is `true`), "Client Location" appears as "Custody Location." Otherwise, it remains as Client Location.
+::: 
+
+## **Components Used**
+
```
Comment (varortz) [inline docs/dev/app/listings.md]: can we add a screenshot here? Lemme send it over!

File: docs/dev/app/listings.md (hunk 18 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: changed!

File: docs/dev/app/listings.md (hunk 22 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: sg, removed!

File: docs/dev/app/listings.md (hunk 23 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: gotcha, so the description would just be plain text like this? 
Screen Shot 2024-06-03 at 2 05 26 PM

do we want to keep the note block about Client/Custody Location?

File: docs/dev/app/listings.md (hunk 24 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: added! Since this screenshot contains an image of ListingCard, do we want to remove the image in the ListingCard section?

File: docs/dev/app/misc.md (hunk 21 above)
Comment (ccatherinetan) [reply docs/dev/app/misc.md]: gotcha, i'll change the page to styling.md as suggested for now, and we can flesh it out on the next PR?

File: docs/dev/app/listings.md (hunk 24 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: removed ListingCard image!

File: docs/dev/app/listings.md (hunk 19 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: added!

File: docs/dev/app/listings.md [hunk 25]
```diff
@@ -1,9 +1,441 @@
...
+- `ListingPage`
+    - `FilterDropdown`
+    - `ListingCard` 
+    - `ListingDetails`
+        - `ProfileMatch`
+        - `InterestForm`
+
+![ListingPage Component Structure](/assets/image/listing_page_components.png)
```
Comment (varortz) [inline docs/dev/app/listings.md]: Could we put the image before the listed components?

File: docs/dev/app/listings.md [hunk 26]
```diff
@@ -1,9 +1,441 @@
...
+
+- `onChange`: **`(newValue: Set<string>) => void`**
+  
+   Usually of the form `newValue => setFieldFilters(newValue)`. This function is called when the values of the filter dropdown is changed, i.e. when the user selects/deselects a value of the current filter.
+
+- (optional) `fullText`: **`string`**
+  
+   A string that is the default display name of the filter dropdown button. Currently, this is only used for the "Remote/In Person" filter.
+
+::: note
```
Comment (varortz) [inline docs/dev/app/listings.md]: let's make this into an info container (`::: info`) so that it can stand out a bit more:
image

Feel free to add a different text other than "info" or keep it as "info", whatever you think is best!

File: docs/dev/app/listings.md [hunk 27]
```diff
@@ -1,9 +1,441 @@
...
+}
+</style>
 
-## Language Support
+| Listing Page | Case | Limited Case Assignments | Language Support |
+|:------------:|------|--------------------------|------------------|
+| Filters | <ol><li>Remote/In Person</li><li>Languages</li><li>Adjudicating Agency</li><li>Country of Origin</li></ol> | <ol><li>Country Field</li><li>Language(s)</li></ol> | <ol><li>Listing Type</li><li>Language(s)</li></ol> | <ol><li>Listing Type</li><li>Language(s)</li></ol> | <ol><li>Listing Type</li><li>Language(s)</li></ol> |
+| Listing Highlights | <ol><li>Relief sought</li><li>Time Commitment</li><li>Remote/In Person</li><li>Adjudicating Agency</li><li>Client Languages</li><li>Client Country of Origin</li><li>__*__ Client Location</li></ol> | <ol><li>Country Field</li><li>Language(s)</li><li>Expected Deliverable</li></ol> | **DOC** <ol><li>Language(s)</li><li>Number of Pages</li></ol> **CASE_INT** <ol><li>Language(s)</li><li>Time Commitment</li><li> Remote/In Person </li></ol> **INT** <ol><li>Language(s) </li><li>Remote/In Person</li></ol> |
+
+::: note (*) Note:
```
Comment (varortz) [inline docs/dev/app/listings.md]: Let's make this an `::: info ...` container so that it can stand out more to the user. I feel like the note container can blend into the background a bit.

File: docs/dev/app/listings.md [hunk 28]
```diff
@@ -1,9 +1,441 @@
 # Listing Pages
 
-Basic structure; feel free to deviate.
+All three listing pages contain a filter bar at the top, a scrollable list of Listing Cards on the left, and the Listing Details of the currently selected listing card on the right. Differences in display between listing pages are listed below. Then, the usage and functionality of components used to bulid the listing pages are explained. 
 
-## Cases
+## Listing Page Comparison
```
Comment (varortz) [inline docs/dev/app/listings.md]: What if we made this comparison section a subsection of the "Components Used" header below? I was thinking the visual of the page and the listed used components could serve as useful context for this comparison table

**Review by varortz — COMMENTED** [review]
Thank you so so so so much for the listing pages docs, massive help! Love how you included tables to make the information more digestable as well. Everything looks great, just left some final comments on some little things :)

File: docs/dev/app/listings.md (hunk 25 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: sg! i moved the image to go in front of the Components Used section

File: docs/dev/app/listings.md (hunk 26 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: sg! changed

File: docs/dev/app/listings.md (hunk 28 above)
Comment (ccatherinetan) [reply docs/dev/app/listings.md]: gotcha! moved!

**Review by varortz — APPROVED** [review]
